tests/Tests.elm

```diff
@@ -72,6 +73,69 @@ expectationTests =
         ]
 
 
+withinTests =
+    describe "Expect.within"
+        [ fuzz float "pythagorean identity" <|
+            \x ->
+                (sin x) ^ 2 + (cos x) ^ 2 |> Expect.within (AbsoluteOrRelative 0.000001 0.00001) 1.0
+        , test "floats known to not add exactly" <|
+            \() -> 0.1 + 0.2 |> Expect.within (Absolute 0.000000001) 0.3
+        , test "approximation of pi" <|
+            \() -> 3.14 |> Expect.within (Absolute 0.01) pi
+        , fuzz2 float float "self equality" <|
+            \epsilon value ->
+                let
+                    eps =
+                        if epsilon /= 0 then
+                            epsilon
+                        else
+                            1
+                in
+                    value |> Expect.within (Relative (abs eps)) value
+        , fuzz float "NaN equality" <|
+            \epsilon ->
+                let
+                    nan =
+                        0.0 / 0.0
+                in
+                    nan |> Expect.notWithin (Relative (abs epsilon)) nan
+        , fuzz float "Infinity equality" <|
+            \epsilon ->
+                let
+                    infinity =
+                        1.0 / 0.0
+                in
+                    infinity |> Expect.within (Relative epsilon) infinity
+        , fuzz float "Negative infinity equality" <|
+            \epsilon ->
+                let
+                    infinity =
+                        -1.0 / 0.0
+                in
+                    infinity |> Expect.within (Relative epsilon) infinity
```

Comment:
This should be named `negative_infinity`.

It might be worth breaking out these constants to the top level.